### PR TITLE
[Tests-Only] Convert fileTrashTest.js to pact.io

### PR DESCRIPTION
**Pull request by swoichha**
Convert `tests/fileTrashTest.js` to pact.io

## Related Issue:
part of #500

### CODE REVIEW

File: tests/fileTrashTest.js [hunk 1]
```diff
@@ -177,18 +463,135 @@ describe('oc.fileTrash', function () {
...
+            headers: {
+              'Content-Type': 'text/html; charset=utf-8',
+              'Access-Control-Allow-Origin': origin,
+              'Access-Control-Allow-Headers': accessControlAllowHeaders,
+              'Access-Control-Allow-Method': accessControlAllowMethods
+            }
+          }
+        }))
+        promises.push(provider.addInteraction({
+          uponReceiving: 'PROPFIND to an empty trashbin after restoring',
```
Comment (individual-it) [inline tests/fileTrashTest.js:494]: looks same like `'PROPFIND to an empty trashbin',` can we use one interaction for both?

File: tests/fileTrashTest.js [hunk 2]
```diff
@@ -207,61 +610,231 @@ describe('oc.fileTrash', function () {
...
+            headers: {
+              'Content-Type': 'text/html; charset=utf-8',
+              'Access-Control-Allow-Origin': origin,
+              'Access-Control-Allow-Headers': accessControlAllowHeaders,
+              'Access-Control-Allow-Method': accessControlAllowMethods
+            }
+          }
+        }))
+        promises.push(provider.addInteraction({
+          uponReceiving: 'PROPFIND trash items after restoring deleting file',
```
Comment (individual-it) [inline tests/fileTrashTest.js:736]: looks same like `'PROPFIND to an empty trashbin',`

File: tests/config/config.drone.json [hunk 3]
```diff
@@ -36,5 +36,7 @@
       "JbdgMGkt3Cq6B0u"
     ],
     "testContent": "123456",
-    "expirationDate": "2022-10-01"
+  "expirationDate": "2022-10-01",
+  "deletedFolderId": "2147596415",
+  "deletedFileId": "2147596419"
```
Comment (individual-it) [inline tests/config/config.drone.json:41]: if those values are only used in `tests/fileTrashTest.js` then lets put them there

**Review by individual-it — CHANGES_REQUESTED** [review]
some interactions look like they are dublicated also you could do some  path more dynamic `.../admin/..`

File: tests/fileTrashTest.js (hunk 1 above)
Comment (swoichha) [reply tests/fileTrashTest.js:494]: Yes, we can use one interaction for both. Need to do some refactoring for that which is WIP.

File: tests/fileTrashTest.js [hunk 4]
```diff
@@ -177,18 +463,135 @@ describe('oc.fileTrash', function () {
...
+            }),
+            headers: {
+              Destination: config.owncloudURL + 'remote.php/dav/files/admin/' + config.testFolder + '%20(restored%20to%20a%20different%20location)'
+            }
+          },
+          willRespondWith: {
+            status: 201,
+            headers: {
+              'Content-Type': 'text/html; charset=utf-8',
+              'Access-Control-Allow-Origin': origin,
```
Comment (jasson99) [inline tests/fileTrashTest.js:487]: we have variable for this

**Review by jasson99 — CHANGES_REQUESTED** [review]

File: tests/fileTrashTest.js [hunk 5]
```diff
@@ -60,106 +124,291 @@ describe('oc.fileTrash', function () {
...
+              '</d:propfind>'
+          },
+          willRespondWith: {
+            status: 207,
+            headers: {
+              'Access-Control-Allow-Origin': origin,
+              'Access-Control-Allow-Headers': accessControlAllowHeaders,
+              'Access-Control-Allow-Methods': accessControlAllowMethods,
+              'Content-Type': 'text/html; charset=utf-8'
+            },
```
Comment (kiranparajuli589) [inline tests/fileTrashTest.js:383]: I think this is also used in multiple interactions! You can give it a name too.

File: tests/fileTrashTest.js [hunk 6]
```diff
@@ -176,20 +425,137 @@ describe('oc.fileTrash', function () {
...
+        promises.push(provider.addInteraction({
+          uponReceiving: 'MOVE folder from trashbin to fileslist to a different location',
+          withRequest: {
+            method: 'MOVE',
+            path: Pact.Matchers.term({
+              matcher: '.*\\/remote\\.php\\/dav\\/trash-bin\\/admin\\/' + config.deletedFolderId,
+              generate: '/remote.php/dav/trash-bin/admin/' + config.deletedFolderId
+            }),
+            headers: {
+              Destination: config.owncloudURL + 'remote.php/dav/files/admin/' + config.testFolder + '%20(restored%20to%20a%20different%20location)'
```
Comment (kiranparajuli589) [inline tests/fileTrashTest.js:443]: you can use a matcher to compare the destination as in `path`

**Review by individual-it — APPROVED** [review]

File: tests/fileTrashTest.js [hunk 7]
```diff
@@ -60,106 +143,228 @@ describe('oc.fileTrash', function () {
...
-      testFolder = 'testFolder' + suffix
-      testFile = testFolder + '/file.txt'
-      oc.files.createFolder(testFolder).then(() => {
-        oc.files.putFileContents(testFile, '*').then(() => {
-          oc.files.delete(testFolder).then(() => {
-            done()
-          })
-        })
-      })
+  fdescribe('and when empty', function () {
```
Comment (jasson99) [inline tests/fileTrashTest.js:167]: You could make it more clear. something like `when the trashbin is empty`

Comment (swoichha) [reply tests/fileTrashTest.js:167]: Yes, but originally it was `and when empty`. So I'm not sure if we should change it or not https://github.com/owncloud/owncloud-sdk/blob/26bc93b166d001835a4995249af0bb207729818c/tests/fileTrashTest.js#L77

**Review by jasson99 — COMMENTED** [review]
if you want all the tests in this file to run, then you can simply let the tests be as `it` instead of `fit`.

**Pull request merged**